src/android/WebAppConfiguration.java

```diff
@@ -62,14 +63,28 @@ public void setLastKnownGoodVersion(String version) {
     }
 
     public Set<String> getBlacklistedVersions() {
-        return preferences.getStringSet("blacklistedVersions", Collections.EMPTY_SET);
+        Set<String> blacklistedVersions = preferences.getStringSet("blacklistedVersions", Collections.EMPTY_SET);
+        Log.d("BLACKLIST", "getBlacklistedVersions: " + blacklistedVersions);
+        return blacklistedVersions;
     }
 
     public void addBlacklistedVersion(String version) {
+        Set<String> versionsForRetry = new HashSet<String>(preferences.getStringSet("versionsForRetry", Collections.EMPTY_SET));
         Set<String> blacklistedVersions = new HashSet<String>(getBlacklistedVersions());
+        Log.d("BLACKLIST", "versionsForRetry: " + versionsForRetry);
+
+        if (!versionsForRetry.contains(version) && !blacklistedVersions.contains(version)) {
+            Log.d("BLACKLIST", "adding faulty version for retry: " + version);
+            versionsForRetry.add(version);
+            preferences.edit().putStringSet("versionsForRetry", versionsForRetry).commit();
+        } else if () {
```

Comment:
typo? This fails with `WebAppConfiguration.java:80: error: illegal start of expression`

Reply:
oh my... thanks for checking it. accidentally pushed from a branch with an intermediate state